modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt

```diff
@@ -136,24 +136,30 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
     ): List<IrohaContainer> = coroutineScope {
         val keyPairs = mutableListOf<KeyPair>()
         val portsList = mutableListOf<List<Int>>()
+
         repeat(withIroha.amount) {
             keyPairs.add(generateKeyPair())
             portsList.add(findFreePorts(3)) // P2P + API + TELEMETRY
         }
-
         val peerIds = keyPairs.mapIndexed { i: Int, kp: KeyPair ->
             val p2pPort = portsList[i][IrohaConfig.P2P_PORT_IDX]
             kp.toPeerId(IrohaContainer.NETWORK_ALIAS + p2pPort, p2pPort)
         }
-
         val deferredSet = mutableSetOf<Deferred<*>>()
         val containers = Collections.synchronizedList(ArrayList<IrohaContainer>(withIroha.amount))
         repeat(withIroha.amount) { n ->
             async(Dispatchers.IO) {
                 val p2pPort = portsList[n][IrohaConfig.P2P_PORT_IDX]
                 val container = IrohaContainer {
                     networkToJoin = network
-                    genesis = withIroha.sources.map { it.createInstance() }.toSingle()
+                    genesisPath = when {
+                        withIroha.source.isEmpty() -> null
+                        else -> withIroha.source
+                    }
+                    genesis = when {
```

Comment:
how about actually creating a genesis just here and get rid of the `genesisPath` in `IrohaConfig`?